presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PackageController.java

```diff
@@ -234,60 +234,92 @@ public Response restart()
 
     /**
      * The status api can be used to ascertain whether presto is installed and running.
-     * If Presto is installed, this method returns a JSON file containing Presto version
-     * If Presto is running, this method returns a JSON file containing the following details
-     * nodeVersion : Presto version installed in the node
-     * environment : The name of the environment
-     * coordinator : Specifies whether the current node functions as a coordinator
-     * state : The state of the node. ACTIVE (or) SHUTTING_DOWN
+     * <p>
+     * This information may be included in a response:
+     * <ul>
+     * <li>{@code installed}: whether Presto is installed
+     *      </li>
+     * <li>{@code version}: what version of Presto appears to be running
+     *      </li>
+     * <li>{@code running}: whether Presto is running
+     *      </li>
+     * <li>{@code state}: the Presto node's state, as reported by
+     *                    "{@code GET {presto}/v1/info/state}"
+     *      </li>
+     * <li>{@code ServerInfo}: Presto server info, as reported by
+     * "                   {@code GET {presto}/v1/info}"
+     *      </li>
+     * <li>{@code coordinator}: An object with two keys:
+     *      <ul>
+     *              <li>{@code running}: Whether the node is running as a coordinator
+     *                   </li>
+     *              <li>{@code configured}: Whether the node is configured
+     *                                      to run as a coordinator
+     *                   </li>
+     *      </ul>
+     *      </li>
+     * </ul>
      *
      * @return Response containing node status
      */
     public Response status()
     {
-        try {
-            if (!isInstalled()) {
-                LOGGER.info("Presto is not installed");
-                return Response.status(OK).entity(ImmutableMap.of("installed", false)).type(APPLICATION_JSON)
+        ImmutableMap.Builder<String, Object> jsonBuilder = ImmutableMap.builder();
+        // Call this to get a return value on success
+        Supplier<Response> responseGetter = () ->
```

Comment:
But you have to define `Supplier` which is not necessary here. You introduce extra dependency which does not serve any purpose. It simply obfuscates the code.